README.md
# joomleague
JoomLeague is a sports league management tool for Joomla!® 4.x.x


com_joomleague/CHANGELOG
commit 57ae969002ca9c620da5ac3da032953e2248654e
Author: And_One <andone@mfga.at>
Date:   Tue Jan 3 09:54:23 2017 +0100

    updated language files (thx to all translators @ https://opentranslators.transifex.com/projects/p/joomleague)

 18 files changed, 1533 insertions(+), 1533 deletions(-)

commit 6398326bc55f8093c7649d70f83762304b61ae53
Author: And_One <andone@mfga.at>
Date:   Thu Sep 29 16:51:16 2016 +0200

    updated semantic version string and download package on http://www.joomleague.at to 3.0.21.124789c-Blue

 32 files changed, 74 insertions(+), 33 deletions(-)

commit 124789cea36383ea4eb7e495e68b963205759225
Author: And_One <andone@mfga.at>
Date:   Thu Sep 29 16:50:33 2016 +0200

    updated language files (thx to all translators @ https://opentranslators.transifex.com/projects/p/joomleague)

 2 files changed, 54 insertions(+), 54 deletions(-)

commit 630c60ae31a072a925f7730b727de41286b8cf4b
Author: And_One <andone@mfga.at>
Date:   Thu Sep 29 14:47:41 2016 +0000

    Update default_selectround.php

 1 file changed, 1 insertion(+), 2 deletions(-)

commit aa33ba65031a89fc29b77a0f6977fe3a7e981b13
Author: And_One <andone@mfga.at>
Date:   Thu Sep 29 14:47:16 2016 +0000

    Update view.html.php

 1 file changed, 17 insertions(+), 3 deletions(-)

commit 6eabd1e51ac9a82eb0c9794c1010b9fcc6bba3e5
Author: Hermann Herz <heart1010@gmail.com>
Date:   Thu Sep 29 14:23:54 2016 +0000

    change td width for events icon from 5 to 16
    http://forum.joomleague.at/viewtopic.php?p=85878#p85878

 1 file changed, 1 insertion(+), 1 deletion(-)

commit 3835843c0e7326c0776fbe565142bca731c4e9b3
Author: And_One <andone@mfga.at>
Date:   Mon Sep 26 22:42:38 2016 +0200

    updated semantic version string and download package on http://www.joomleague.at to 3.0.20.6903c5f-Blue

 32 files changed, 41 insertions(+), 33 deletions(-)

commit 6903c5ff832d52a46bc9d0814c45065d115fae53
Author: And_One <andone@mfga.at>
Date:   Mon Sep 26 22:32:29 2016 +0200

    updated semantic version string and download package on http://www.joomleague.at to 3.0.19.271d82d-Blue

 32 files changed, 51 insertions(+), 33 deletions(-)

commit 271d82da32cae83b60cbabe0931b72cfa2304ed9
Author: And_One <andone@mfga.at>
Date:   Mon Sep 26 22:29:59 2016 +0200

    DEV: fixed call to undefined method in Backend
    /administrator/components/com_joomleage/views/joomleague/view_html.php
    Line 232

 6 files changed, 13 insertions(+), 25 deletions(-)

commit 35b7d0745c2994419063acfee3c58a53fa445c92
Author: And_One <andone@mfga.at>
Date:   Mon Sep 26 10:35:05 2016 +0200

    updated semantic version string and download package on http://www.joomleague.at to 3.0.18.31e497f-Blue

 32 files changed, 49 insertions(+), 33 deletions(-)

commit 31e497f337cb8cb075e19698d853c6dd3ec66ce3
Author: Hermann Herz <heart1010@gmail.com>
Date:   Mon Sep 26 07:58:32 2016 +0000

    wrong css class for select option fields, see http://forum.joomleague.at/viewtopic.php?f=161&t=19109&p=85854#p85854

 1 file changed, 2 insertions(+), 4 deletions(-)

commit 16b805a510cc1c165b0ec2c5795e508dda474a55
Author: And_One <andone@mfga.at>
Date:   Mon Sep 26 09:51:04 2016 +0200

    updated semantic version string and download package on http://www.joomleague.at to 3.0.17.fe8d736-Blue

 32 files changed, 57 insertions(+), 33 deletions(-)

commit fe8d736d1e24e53dbe8aabec91b66d0d1bb4a173
Author: And_One <andone@mfga.at>
Date:   Mon Sep 26 09:50:11 2016 +0200

    updated language files (thx to all translators @ https://opentranslators.transifex.com/projects/p/joomleague)

 4 files changed, 370 insertions(+), 370 deletions(-)

commit 52abe62713643750548e58f14e0dd6b10c4427a3
Author: And_One <andone@mfga.at>
Date:   Tue Sep 20 21:18:42 2016 +0200

    DEV: updated copyright string to 2016

 122 files changed, 153 insertions(+), 153 deletions(-)

commit 5f00e511588a03f0168f889b2515b3b22f94136b
Author: And_One <andone@mfga.at>
Date:   Tue Sep 20 20:38:25 2016 +0200

    updated semantic version string and download package on http://www.joomleague.at to 3.0.16.5739060-Blue

 32 files changed, 49 insertions(+), 33 deletions(-)

commit 5739060bacecd9f8be7871df78af8cbfd962ff6f
Author: And_One <andone@mfga.at>
Date:   Tue Sep 20 20:32:53 2016 +0200

    FIXED results view  rounds dropdown

 1 file changed, 6 insertions(+), 9 deletions(-)

commit e9942b871f5ba95a53d38ed9c74c4de60b29d3eb
Author: And_One <andone@mfga.at>
Date:   Tue Sep 20 16:06:38 2016 +0200

    updated semantic version string and download package on http://www.joomleague.at to 3.0.15.f609ab0-Blue

 32 files changed, 65 insertions(+), 33 deletions(-)

commit f609ab02f87f88390d9dc314bd24e3c5054cf3db
Author: And_One <andone@mfga.at>
Date:   Tue Sep 20 16:06:14 2016 +0200

    updated language files (thx to all translators @ https://opentranslators.transifex.com/projects/p/joomleague)

 1 file changed, 131 insertions(+), 131 deletions(-)

commit 4dea0af99275cac26adec787821f0ac4accbff40
Author: And_One <andone@mfga.at>
Date:   Tue Sep 20 14:02:42 2016 +0000

    Update match.php

 1 file changed, 3 insertions(+), 1 deletion(-)

commit 6de4fd2051dd4afeaca2f4a1104ae2e751367733
Author: And_One <andone@mfga.at>
Date:   Mon Sep 19 13:03:12 2016 +0000

    Update jl_install.php

 1 file changed, 2 insertions(+), 2 deletions(-)

commit 1649d2037d9de475db98435fefc36aba8ea5c2dc
Author: And_One <andone@mfga.at>
Date:   Mon Sep 19 09:32:25 2016 +0200

    updated semantic version string and download package on http://www.joomleague.at to 3.0.14.d74d897-Blue

 32 files changed, 48 insertions(+), 32 deletions(-)

commit d74d8974ef85d47353ae974625bb3567cd2649af
Author: And_One <andone@mfga.at>
Date:   Mon Sep 19 09:31:35 2016 +0200

    updated language files (thx to all translators @ https://opentranslators.transifex.com/projects/p/joomleague)

 3 files changed, 337 insertions(+), 337 deletions(-)

commit a133f3f8a497898834bed16ff7324f681145d224
Author: And_One <andone@mfga.at>
Date:   Thu Sep 15 17:27:42 2016 +0200


com_joomleague/components/com_joomleague/helpers/html.php
<?php
/**
 * @copyright	Copyright (C) 2006-2014 joomleague.at. All rights reserved.
 * @license		GNU/GPL,see LICENSE.php
 * Joomla! is free software. This version may have been modified pursuant
 * to the GNU General Public License,and as distributed it includes or
 * is derivative of works licensed under the GNU General Public License or
 * other free or open source software licenses.
 * See COPYRIGHT.php for copyright notices and details.
 */

// Check to ensure this file is included in Joomla!
use Joomla\CMS\Factory;
use Joomla\CMS\Table\Table;
use Joomla\CMS\Uri\Uri;
use Joomla\CMS\MVC\Model\BaseDatabaseModel;

defined('_JEXEC') or die;

/**
 * provides html code snippets for the views
 * @author julienV
 */
class JoomleagueHelperHtml {


	/**
	 * Return formatted match time
	 *
	 * @param object $game
	 * @param array $config
	 * @param array $overallconfig
	 * @param object $project
	 * @return string html
	 */
	public static function showMatchTime(&$game, &$config, &$overallconfig, &$project)
	// overallconfig could be deleted here and replaced below by config as both array were merged in view.html.php
	{
		$output='';

		if (!isset($overallconfig['time_format'])) {
			$overallconfig['time_format']='H:i';
		}
		$timeSuffix=JText::_('COM_JOOMLEAGUE_GLOBAL_CLOCK');
		if ($timeSuffix=='COM_JOOMLEAGUE_GLOBAL_CLOCK') {
			$timeSuffix='%1$s&nbsp;h';
		}

		if ($game->match_date)
		{
			$matchTime = JoomleagueHelper::getMatchTime($game, $overallconfig['time_format']);

			if ($config['show_time_suffix'] == 1)
			{
				$output .= sprintf($timeSuffix,$matchTime);
			}
			else
			{
				$output .= $matchTime;
			}

			$config['mark_now_playing']=(isset($config['mark_now_playing'])) ? $config['mark_now_playing'] : 0;

			if ($config['mark_now_playing'])
			{
				$totalMatchDuration = ($project->halftime * ($project->game_parts - 1)) + $project->game_regular_time;
				if ($project->allow_add_time == 1 && ($game->team1_result == $game->team2_result))
				{
					$totalMatchDuration += $project->add_time;
				}
				$project_tz = new DateTimeZone($game->timezone);
				$startTimestamp = JoomleagueHelper::getMatchStartTimestamp($game);
				$startTime = new DateTime($startTimestamp, $project_tz);
				$endTime = new DateTime($startTimestamp, $project_tz);
				$endTime->add(new DateInterval('PT'.$totalMatchDuration.'M'));
				$now = new DateTime('now', $project_tz);
				if ($now >= $startTime && $now <= $endTime)
				{
					$match_begin=$output.' ';
					$title=str_replace('%STARTTIME%',$match_begin,trim(htmlspecialchars($config['mark_now_playing_alt_text'])));
					$title=str_replace('%ACTUALTIME%',self::mark_now_playing($thistime,$match_stamp,$config,$project),$title);
					$styletext='';
					if (isset($config['mark_now_playing_blink']) && $config['mark_now_playing_blink'])
					{
						$styletext=' style="text-decoration:blink"';
					}
					$output='<b><i><acronym title="'.$title.'"'.$styletext.'>';
					$output .= $config['mark_now_playing_text'];
					$output .= '</acronym></i></b>';
				}
			}
		}
		else
		{
			$matchTime='--&nbsp;:&nbsp;--';
			if ($config['show_time_suffix'])
			{
				$output .= sprintf($timeSuffix,$matchTime);
			}
			else
			{
				$output .= $matchTime;
			}
		}

		return $output;
	}

	/**
	 * prints teams names and divisions...
	 *
	 * @param int $projectId
	 * @param object $homeTeam
	 * @param object $guestTeam
	 * @param array $config
	 * @return string html
	 */
	public static function showDivisonRemark($projectId, &$homeTeam, &$guestTeam, &$config)
	{
		$output='';
		if ($config['switch_home_guest'])
		{
			$tmpteam = $homeTeam;
			$homeTeam = $guestTeam;
			$guestTeam = $tmpteam;
		}
		if (isset($homeTeam) && $homeTeam->division_id > 0 && isset($guestTeam) && $guestTeam->division_id > 0)
		{
			//TODO: Where is spacer defined???
			if (!isset($config['spacer']))
			{
				$config['spacer']='/';
			}

			$nametype = 'division_' . $config['show_division_name'];

			if ($config['show_division_link'])
			{
				$link = JoomleagueHelperRoute::getRankingRoute($projectId, null, null, null, 0, $homeTeam->division_id);
				$output .= JHtml::link($link, $homeTeam->$nametype);
			}
			else
			{
				$output .= $homeTeam->$nametype;
			}

			if ($homeTeam->division_id != $guestTeam->division_id)
			{
				$output .= $config['spacer'];
				if ($config['show_division_link'] == 1)
				{
					$link = JoomleagueHelperRoute::getRankingRoute($projectId, null, null, null, 0, $guestTeam->division_id);
					$output .= JHtml::link($link, $guestTeam->$nametype);
				}
				else
				{
					$output .= $guestTeam->$nametype;
				}
			}
		}
		else
		{
			$output .= '&nbsp;';
		}
		return $output;
	}

	/**
	 * Shows matchday title
	 *
	 * @param string $title
	 * @param int $current_round
	 * @param array $config
	 * @param int $mode
	 * @return string html
	 */
	public static function showMatchdaysTitle($title, $current_round, &$config, $mode=0)
	{
		$app 	= Factory::getApplication();
		$jinput = $app->input;
		
		$projectid = $jinput->getInt('p',0);
		
		$mdlProject = BaseDatabaseModel::getInstance('project','JoomleagueModel');
		$mdlProject->setProjectID($projectid);
		$project = $mdlProject->getProject();
		
		echo ($title != '') ? $title.' - ' : $title;
		if ($current_round > 0)
		{
			$thisround = Table::getInstance('Round','Table');
			$thisround->load($current_round);

			if ($config['type_section_heading'] == 1 && $thisround->name != '')
			{
				if ($mode == 1)
				{
					$link=JoomleagueHelperRoute::getRankingRoute($projectid,$thisround->id);
					echo JHtml::link($link,$thisround->name);
				}
				else
				{
					echo $thisround->name;
				}
			}
			elseif ($thisround->roundcode > 0)
			{
				echo ' '.JText::sprintf('COM_JOOMLEAGUE_RESULTS_MATCHDAY', $thisround->roundcode).'&nbsp;';
			}

			if ($config['show_rounds_dates'] == 1)
			{
				echo " (";
				if (! strstr($thisround->round_date_first,"0000-00-00"))
				{
					echo JHtml::date($thisround->round_date_first .' UTC',
										'COM_JOOMLEAGUE_GLOBAL_CALENDAR_DATE',
										JoomleagueHelper::getTimezone($project,$config));
				}
				if (($thisround->round_date_last != $thisround->round_date_first) &&
				(! strstr($thisround->round_date_last,"0000-00-00")))
				{
					echo " - ".JHtml::date($thisround->round_date_last .' UTC',
											'COM_JOOMLEAGUE_GLOBAL_CALENDAR_DATE',
											JoomleagueHelper::getTimezone($project,$config));
				}
				echo ")";
			}
		}
	}

	/**
	 * display match playground
	 *
	 * @param int $projectId
	 * @param array $teams
	 * @param object $game
	 * @param array $config
	 * @return string
	 */
	public static function showMatchPlayground($projectId, $teams, &$game, $config)
	{
		if (($config['show_playground'] || $config['show_playground_alert']) && isset($game->playground_id))
		{
			if (empty($game->playground_id))
			{
				$game->playground_id = $teams[$game->projectteam1_id]->standard_playground;
			}
			if (empty($game->playground_id))
			{
				$cinfo = Table::getInstance('Club','Table');
				$cinfo->load($teams[$game->projectteam1_id]->club_id);
				$game->playground_id = $cinfo->standard_playground;
				$teams[$game->projectteam1_id]->standard_playground = $cinfo->standard_playground;
			}

			if (!$config['show_playground'] && $config['show_playground_alert'])
			{
				if ($teams[$game->projectteam1_id]->standard_playground == $game->playground_id)
				{
					echo '-';
					return '';
				}
			}

			$boldStart	= '';
			$boldEnd	= '';
			$toolTipTitle	= JText::_('COM_JOOMLEAGUE_PLAYGROUND_MATCH');
			$toolTipText	= '';

			if ($config['show_playground_alert'] &&
				$teams[$game->projectteam1_id]->standard_playground != $game->playground_id)
			{
				$boldStart		= '<b style="color:red; ">';
				$boldEnd		= '</b>';
				$toolTipTitle	= JText::_('COM_JOOMLEAGUE_PLAYGROUND_NEW');
			}

			$pginfo = Table::getInstance('Playground','Table');
			$pginfo->load($game->playground_id);

			$toolTipText	.= $pginfo->name . '&lt;br /&gt;';
			$toolTipText	.= $pginfo->address . '&lt;br /&gt;';
			$toolTipText	.= $pginfo->zipcode . ' ' . $pginfo->city . '&lt;br /&gt;';

			$link = JoomleagueHelperRoute::getPlaygroundRoute($projectId,$game->playground_id);
			$playgroundName = $config['show_playground_name'] == 'name' ? $pginfo->name : $pginfo->short_name;
			?>
		<span class='hasTip' title='<?php echo $toolTipTitle; ?> :: <?php echo $toolTipText; ?>'>
			<?php	echo JHtml::link($link, $boldStart . $playgroundName . $boldEnd); ?>
		</span>
	<?php
		}
	}

	/**
	 * mark currently playing game
	 *
	 * @param int $thistime
	 * @param int $match_stamp
	 * @param array $config
	 * @param object $project
	 * @return string
	 */
	public function mark_now_playing($thistime,$match_stamp,&$config,&$project)
	{
		$whichpart=1;
		$gone_since_begin=intval(($thistime - $match_stamp)/60);
		$parts_time=intval($project->game_regular_time / $project->game_parts);
		if ($project->allow_add_time) {
			$overtime=1;
		}else{$overtime=0;
		}
		$temptext=JText::_('COM_JOOMLEAGUE_RESULTS_LIVE_WRONG');
		for ($temp_count=1; $temp_count <= $project->game_parts+$overtime; $temp_count++)
		{
			$this_part_start=(($temp_count-1) * ($project->halftime + $parts_time));
			$this_part_end=$this_part_start + $parts_time;
			$next_part_start=$this_part_end + $project->halftime;
			if ($gone_since_begin >= $this_part_start && $gone_since_begin <= $this_part_end)
			{
				$temptext=str_replace('%PART%',$temp_count,trim(htmlspecialchars($config['mark_now_playing_alt_actual_time'])));
				$temptext=str_replace('%MINUTE%',($gone_since_begin+1 - ($temp_count-1)*$project->halftime),$temptext);
				break;
			}
			elseif ($gone_since_begin > $this_part_end && $gone_since_begin < $next_part_start)
			{
				$temptext=str_replace('%PART%',$temp_count,trim(htmlspecialchars($config['mark_now_playing_alt_actual_break'])));
				break;
			}
		}
		return $temptext;
	}

	/**
	 * return thumb up/down image url if team won/loss
	 *
	 * @param object $game
	 * @param int $projectteam_id
	 * @param array attributes
	 * @return string image html code
	 */
	public static function getThumbUpDownImg($game, $projectteam_id, $attributes = null)
	{
		$res = JoomleagueFrontHelper::getTeamMatchResult($game, $projectteam_id);
		if ($res === false) {
			return false;
		}

		if ($res == 0)
		{
			$img = 'media/com_joomleague/jl_images/draw.png';
			$alt = JText::_('COM_JOOMLEAGUE_GLOBAL_DRAW');
			$title = $alt;
		}
		else if ($res < 0)
		{
			$img = 'media/com_joomleague/jl_images/thumbs_down.png';
			$alt = JText::_('COM_JOOMLEAGUE_GLOBAL_LOST');
			$title = $alt;
		}
		else
		{
			$img = 'media/com_joomleague/jl_images/thumbs_up.png';
			$alt = JText::_('COM_JOOMLEAGUE_GLOBAL_WON');
			$title = $alt;
		}

		// default title attribute, if not specified in passed attributes
		$def_attribs = array('title' => $title);
		if ($attributes) {
			$attributes = array_merge($def_attribs, $attributes);
		}
		else {
			$attributes = $def_attribs;
		}

		return JHtml::image($img, $alt, $attributes);
	}

	/**
	* return thumb up/down image as link with score as title
	*
	* @param object $game
	* @param int $projectteam_id
	* @param array attributes
	* @return string linked image html code
	*/
	public static function getThumbScore($game, $projectteam_id, $attributes = null)
	{
		if (!$img = self::getThumbUpDownImg($game, $projectteam_id, $attributes = null)) {
			return false;
		}
		$txt = $teams[$game->projectteam1_id]->name.' - '.$teams[$game->projectteam2_id]->name.' '.$game->team1_result.' - '. $game->team2_result;

		$attribs = array('title' => $txt);
		if (is_array($attributes)) {
			$attribs = array_merge($attributes, $attribs);
		}
		$url = JRoute::_(JoomleagueHelperRoute::getMatchReportRoute($game->project_slug, $game->slug));
		return JHtml::link($url, $img);
	}

	/**
	* return up/down image for ranking
	*
	* @param object $team (rank)
	* @param object $previous (rank)
	* @param int $ptid
	* @return string image html code
	*/
	public static function getLastRankImg($team,$previous,$ptid,$attributes = null)
	{
		if ( isset( $previous[$ptid]->rank ) )
		{
			$imgsrc = 'media/com_joomleague/jl_images/';
			if ( ( $team->rank == $previous[$ptid]->rank ) || ( $previous[$ptid]->rank == "" ) )
			{
				$imgsrc .= "same.png";
				$alt	 = JText::_('COM_JOOMLEAGUE_RANKING_SAME');
				$title	 = $alt;
			}
			elseif ( $team->rank < $previous[$ptid]->rank )
			{
				$imgsrc .= "up.png";
				$alt	 = JText::_('COM_JOOMLEAGUE_RANKING_UP');
				$title	 = $alt;
			}
			elseif ( $team->rank > $previous[$ptid]->rank )
			{
				$imgsrc .= "down.png";
				$alt	 = JText::_('COM_JOOMLEAGUE_RANKING_DOWN');
				$title	 = $alt;
			}

		$def_attribs = array('title' => $title);
		if ($attributes) {
			$attributes = array_merge($def_attribs, $attributes);
		}
		else {
			$attributes = $def_attribs;
		}
		return JHtml::image($imgsrc,$alt,$attributes);
		}

	}

	public static function printColumnHeadingSort( $columnTitle, $paramName, $config = null, $default="DESC" )
	{
	    $app 	= Factory::getApplication();
		$output = "";
		$img='';
		if ( $config['column_sorting'] || $config == null)
		{
			$params = array(
					"option" => "com_joomleague",
					"view"   => $app->input->getVar("view", "ranking"),
					"p" => $app->input->getInt( "p", 0 ),
					"r" => $app->input->getInt( "r", 0 ),
					"type" => $app->input->getVar( "type", "" ) );

			if ( $app->input->getVar( 'order', '' ) == $paramName )
			{
				$params["order"] = $paramName;
				$params["dir"] = ( $app->input->getVar( 'dir', '') == 'ASC' ) ? 'DESC' : 'ASC';
				$imgname = 'sort'.($app->input->getVar( 'dir', '') == 'ASC' ? "02" :"01" ).'.gif';
				$img = JHtml::image(
										'media/com_joomleague/jl_images/' . $imgname,
				$params["dir"] );
			}
			else
			{
				$params["order"] = $paramName;
				$params["dir"] = $default;
			}
			$query = Uri::buildQuery( $params );
			echo JHtml::link(
			JRoute::_( "index.php?".$query ),
			JText::_($columnTitle),
			array( "class" => "jl_rankingheader" ) ).$img;
		}
		else
		{
			echo JText::_($columnTitle);
		}
	}

	public static function nextLastPages( $url, $text, $maxentries, $limitstart = 0, $limit = 10 )
	{
		$latestlimitstart = 0;
		if ( intval( $limitstart - $limit ) > 0 )
		{
			$latestlimitstart = intval( $limitstart - $limit );
		}
		$nextlimitstart = 0;
		if ( ( $limitstart + $limit ) < $maxentries )
		{
			$nextlimitstart = $limitstart + $limit;
		}
		$lastlimitstart = ( $maxentries - ( $maxentries % $limit ) );
		if ( ( $maxentries % $limit ) == 0 )
		{
			$lastlimitstart = ( $maxentries - ( $maxentries % $limit ) - $limit );
		}

		echo '<center>';
		echo '<table style="width: 50%; align: center;" cellspacing="0" cellpadding="0" border="0">';
		echo '<tr>';
		echo '<td style="width: 10%; text-align: left;" nowrap="nowrap">';
		if ( $limitstart > 0 )
		{
			$query = Uri::buildQuery(
			array(
					"limit" => $limit,
					"limitstart" => 0 ) );
			echo JHtml::link( $url.$query, '&lt;&lt;&lt;' );
			echo '&nbsp;&nbsp;&nbsp';
			$query = Uri::buildQuery(
			array(
					"limit" => $limit,
					"limitstart" => $latestlimitstart ) );
			echo JHtml::link( $url.$query, '&lt;&lt;' );
			echo '&nbsp;&nbsp;&nbsp;';
		}
		echo '</td>';
		echo '<td style="text-align: center;" nowrap="nowrap">';
		$players_to = $maxentries;
		if ( ( $limitstart + $limit ) < $maxentries )
		{
			$players_to = ( $limitstart + $limit );
		}
		echo sprintf( $text, $maxentries, ($limitstart+1).' - '.$players_to );
		echo '</td>';
		echo '<td style="width: 10%; text-align: right;" nowrap="nowrap">';
		if ( $nextlimitstart > 0 )
		{
			echo '&nbsp;&nbsp;&nbsp;';
			$query = Uri::buildQuery(
			array(
					"limit" => $limit,
					"limitstart" => $nextlimitstart ) );
			echo JHtml::link( $url.$query, '&gt;&gt;' );
			echo '&nbsp;&nbsp;&nbsp';
			$query = Uri::buildQuery(
			array(
					"limit" => $limit,
					"limitstart" => $lastlimitstart ) );
			echo JHtml::link( $url.$query, '&gt;&gt;&gt;' );
		}
		echo '</td>';
		echo '</tr>';
		echo '</table>';
		echo '</center>';
	}

}